server/index.js

```diff
@@ -0,0 +1,115 @@
+// app dependencies
+const express = require('express');
+const bodyParser = require('body-parser');
+const expressHandlebars = require('express-handlebars');
+const Handlebars = require('handlebars');
+const helmet = require('helmet');
+const cookieSession = require('cookie-session');
+const http = require('http');
+const logger = require('winston');
+const requestLogger = require('middleware/requestLogger.js');
+const Path = require('path');
+const loggerConfig = require('loggerConfig.js');
+const mysqlConfig = require('mysqlConfig.js');
+const siteConfig = require('siteConfig.js');
+const slackConfig = require('slackConfig.js');
+
+function Server () {
+  this.configureLogger = (userConfig) => {
+    loggerConfig.update(userConfig);
+  };
+  this.configureMysql = (userConfig) => {
+    mysqlConfig.update(userConfig);
+  };
+  this.configureSite = (userConfig) => {
+    siteConfig.update(userConfig);
+  };
+  this.configureSlack = (userConfig) => {
+    slackConfig.update(userConfig);
+  };
+  this.configureModels = () => {
+    logger.debug('here is where you could add/overwrite the default models')
+  };
+  this.configureRoutes = () => {
+    logger.debug('here is where you could add/overwrite the default routes')
+  };
+  this.createApp = () => {
+    // create an Express application
+    const app = express();
+
+    // trust the proxy to get ip address for us
+    app.enable('trust proxy');
+
+    /* add middleware */
+    // set HTTP headers to protect against well-known web vulnerabilties
+    app.use(helmet());
+    // 'express.static' to serve static files from public directory
+    if (siteConfig.routes.publicFolder) {
+      // take in a different public folder, so it can serve it's own bundle if needed
+      const publicFolder = Path.resolve(process.cwd(), siteConfig.routes.publicFolder);
+      app.use(express.static(publicFolder));
+      logger.info('serving static files from custom path:', publicFolder);
+    } else {
+      const publicPath = Path.resolve(process.cwd(), 'public');
+      app.use(express.static(publicPath));
+      logger.warn(`serving static files from default static path at ${publicPath}.  Please specify a path in your config/siteConfig.js file`, );
+    };
+    // 'body parser' for parsing application/json
+    app.use(bodyParser.json());
+    // 'body parser' for parsing application/x-www-form-urlencoded
+    app.use(bodyParser.urlencoded({ extended: true }));
+
+    // add custom middleware (note: build out to accept dynamically use what is in server/middleware/
+    app.use(requestLogger);
+
+    // configure passport
+    const speechPassport = require('speechPassport');
+    // initialize passport
+    const sessionKey = siteConfig.auth.sessionKey;
+    app.use(cookieSession({
+      name  : 'session',
+      keys  : [sessionKey],
+      maxAge: 24 * 60 * 60 * 1000, // i.e. 24 hours
```

Comment:
Do we need to have any `maxAge`? It would be nice if the user was never logged out

Reply:
I guess we don't need to, and it does make a better user experience if the user doesn't need to log back in ever day.  I eliminated the maxAge so sessions should last indefinitely.